syntax = "proto3";

package chasky;

import "chasky/core/framework/function.proto";
import "chasky/core/framework/attr_value.proto";

message GraphDef {
  // Name of the Graph
  string name = 1;

  repeated NodeDef nodes = 2;
}

message NodeDef {
  // The name given to this layer
  string name = 1;

  bool is_source = 2;

  bool is_end = 3;

  // The function's type
  string func = 4;

  // Each input is "node:src_ouput" with "node" being a string name and
  // src_output being the name of node's output
  repeated string input = 6;

  // mixed node's and functioin' attributes
  map<string, AttrValue> attr = 7;
}

